File: amazia-core/src/main/java/com/example/inquiry/controller/GetUnreadInquiryCountController.java
package com.example.inquiry.controller;

import com.example.inquiry.service.GetUnreadInquiryCountService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

/**
 * Console ベルマーク用の未対応件数取得（フェーズ18）。
 *
 * <p>{@code GET /api/console/inquiries/unread-count} → {@code { count: number }}。
 * 設計書 §6.3 真実の元（{@code inquiries.status='NEW'} の COUNT）。
 */
@RestController
@RequestMapping("/api/console/inquiries")
public class GetUnreadInquiryCountController {

    private final GetUnreadInquiryCountService service;

    public GetUnreadInquiryCountController(GetUnreadInquiryCountService service) {
        this.service = service;
    }

    @GetMapping("/unread-count")
    public ResponseEntity<Map<String, Long>> handle() {
        return ResponseEntity.ok(Map.of("count", service.count()));
    }
}

File: amazia-console/app/Inquiry/Controller/GetUnreadInquiryCountController.php
<?php

namespace App\Inquiry\Controller;

use App\Http\Controllers\Controller;
use App\Inquiry\Service\GetUnreadInquiryCountService;
use Illuminate\Http\Request;

/**
 * フェーズ18: ベルマーク用未対応件数 Controller。
 * GET /api/console/inquiries/unread-count
 */
class GetUnreadInquiryCountController extends Controller
{
    private GetUnreadInquiryCountService $service;

    public function __construct(GetUnreadInquiryCountService $service)
    {
        $this->service = $service;
    }

    public function __invoke(Request $request)
    {
        $userId = $request->input('auth_user_id');
        $response = $this->service->get($userId);
        return response()->json($response->json(), $response->status());
    }
}
